components/skus/browser/rs/lib/src/sdk/credentials/mod.rs

```diff
@@ -37,6 +37,15 @@ where
             return Err(InternalError::OrderLocationMismatch.into());
         }
 
+        // if we have no credentials at all, we should attempt to
+        // fetch_order_credentials to populate prior to trying to 
+        // presenting the order cred summary to the caller, otherwise we 
+        // will get an InternalError::NotFound
+        if !self.client.has_credentials(&order.id).await? {
+            self.fetch_order_credentials(order_id).await?;
```

Comment:
Can this still fail the same way? Just curious if we should actually catch/handle the error (along with this change)

Reply:
Yes, potentially it can still fail the same way.  The root of the problem is the below:
https://github.com/brave/brave-core/blob/bh-skus-sdk-fetch-creds-if-absent-summary/components/skus/browser/rs/cxx/src/lib.rs#L476-L477

Returning an empty string "" as a summary is breaking the below:
https://github.com/brave/brave-core/blob/bh-skus-sdk-fetch-creds-if-absent-summary/components/skus/renderer/skus_js_handler.cc#L291-L300

I would maybe suggest we change this to bubble up the underlying skus sdk error to the JS value like we do in the WASM bindings, or pass back an empty object.  In the accounts reference site implementation it is performing a check on if there are credentials for a given domain first in order to check if it needs to perform a provision.

Basically without performing an initial provision call on the skus sdk, any JS code calling credential_summary will fail, this change is attempting to intercept this condition and make a best effort to fetch order credentials first.

We could fix the error propogation, or change the `result_v` check in the renderer to possibly send back an empty object which will allow the JS code to handle the issue appropriately.

Reply:
Pushed a fix for empty string

Reply:
hmm - either my change or this change is breaking unit tests. Taking a look at this